lang/cs/Org.Apache.REEF.IO/PartitionedData/FileSystem/FileSystemInputPartition.cs

```diff
@@ -123,23 +131,23 @@ private ISet<string> Download()
         /// provided by the Serializer
         /// </summary>
         /// <returns></returns>
-        public T GetPartitionHandle()
+        public IEnumerable<T> GetPartitionHandle()
         {
             lock (_lock)
             {
                 if (_copyToLocal)
                 {
                     if (!_localFiles.IsPresent())
                     {
-                        Cache();
+                        Cache(CacheLevel.Disk);
                     }
 
                     // For now, assume IFileDeSerializer is local.
-                    return _fileSerializer.Deserialize(_localFiles.Value);
+                    return new List<T> { _fileSerializer.Deserialize(_localFiles.Value) };
```

Comment:
How do you guarantee the type returned from Deserialize can be converted to IList<T>